src/averageAges.js

```diff
@@ -1,61 +1,100 @@
- *
- * @return {number}
- */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const arrayMen = people.filter(item => item.sex === 'm');
+
+  const arrayAge = arrayMen.map(item => {
+    const age = item.died - item.born;
+
+    return age;
+  });
+
+  const sumaResult = arrayAge.reduce((acc, current) => {
+    return acc + current;
+  }, 0);
+
+  const ageWithCentury = [...arrayMen].filter(item => {
+    const centuryDied = Math.ceil(item.died / 100);
+
+    return century === centuryDied;
+  });
+
+  const sumAgeWithCentury = ageWithCentury.reduce((acc, current) => {
+    const age = current.died - current.born;
+
+    return acc + age;
+  }, 0);
+
+  const averageAge = sumaResult / arrayMen.length;
+  const averageAgeCentury = sumAgeWithCentury / ageWithCentury.length;
+
+  const result = century ? averageAgeCentury : averageAge;
+
+  return result;
 }
 
-/**
- * Implement calculateWomenAverageAge function
- *
- * Function returns average age of women in array. If `withChildren` is
- * specified then function calculates average age only for women with children
- *
- * Hint: To check if a woman has children you should find someone who mention
- * her as mother.
- *
- * @param {object[]} people
- * @param {boolean} withChildren - optional
- *
- * @return {number}
- */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const arrayWomens = people.filter(item => item.sex === 'f');
+
+  const ageAllWomens = arrayWomens.map(item => item.died - item.born);
+  const sumAgeAllWomens = ageAllWomens.reduce(
+    (acc, current) => acc + current, 0);
+  const averageAgeAllWomens = sumAgeAllWomens / arrayWomens.length;
+
+  const arrPeopleMother = people.map(item => item.mother);
+  const arrayWomensWithChild = arrayWomens.filter(
+    item => arrPeopleMother.includes(item.name));
+  const ageWomensWithChild = arrayWomensWithChild.map(
+    item => item.died - item.born);
+  const sumAgeWomenWithChild = ageWomensWithChild.reduce(
+    (acum, item) => acum + item, 0);
+  const averageAgeWomensWithChild = sumAgeWomenWithChild
+  / arrayWomensWithChild.length;
+
+  const result = withChildren ? averageAgeWomensWithChild : averageAgeAllWomens;
+
+  return result;
 }
 
-/**
- * Implement calculateAverageAgeDiff function.
- *
- * The function returns an average age difference between a child and his or her
- * mother in the array. (A mother's age at child birth)
- *
- * If `onlyWithSon` is specified then function calculates age difference only
- * for sons and their mothers.
- *
- * @param {object[]} people
- * @param {boolean} onlyWithSon - optional
- *
- * @return {number}
- */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const arrNamePeople = people.map(item => {
```

Comment:
The `arrNamePeople` array is not necessary. You can use the `some` method to check if a person has a mother in the array.